host/schedule_test.go

```diff
@@ -0,0 +1,541 @@
+	sdkClient, err := sdkclient.Dial(sdkclient.Options{
+		HostPort:      s.hostPort,
+		Namespace:     s.namespace,
+		DataConverter: s.dataConverter,
+	})
+	if err != nil {
+		s.Logger.Fatal("Error when creating SDK client", tag.Error(err))
+	}
+	s.sdkClient = sdkClient
+	s.taskQueue = s.randomizeStr("tq")
+	s.worker = worker.New(s.sdkClient, s.taskQueue, worker.Options{})
+	if err := s.worker.Start(); err != nil {
+		s.Logger.Fatal("Error when starting worker", tag.Error(err))
+	}
+}
+
+func (s *scheduleIntegrationSuite) TearDownTest() {
+	s.worker.Stop()
+	s.sdkClient.Close()
+}
+
+func (s *scheduleIntegrationSuite) TestBasics() {
+	// sqlite tests are run without elasticsearch
+	if s.esClient == nil {
+		s.T().Skip()
+	}
+
+	sid := "sched-test-basics"
+	wid := "sched-test-basics-wf"
+	wt := "sched-test-basics-wt"
+	wt2 := "sched-test-basics-wt2"
+
+	// switch this to test with search attribute mapper:
+	// csa := "AliasForCustomKeywordField"
+	csa := "CustomKeywordField"
+
+	wfMemo := payload.EncodeString("workflow memo")
+	wfSAValue := payload.EncodeString("workflow sa value")
+	schMemo := payload.EncodeString("schedule memo")
+	schSAValue := payload.EncodeString("schedule sa value")
+
+	schedule := &schedulepb.Schedule{
+		Spec: &schedulepb.ScheduleSpec{
+			Interval: []*schedulepb.IntervalSpec{
+				{Interval: timestamp.DurationPtr(3 * time.Second)},
+			},
+		},
+		Action: &schedulepb.ScheduleAction{
+			Action: &schedulepb.ScheduleAction_StartWorkflow{
+				StartWorkflow: &workflowpb.NewWorkflowExecutionInfo{
+					WorkflowId:   wid,
+					WorkflowType: &commonpb.WorkflowType{Name: wt},
+					TaskQueue:    &taskqueuepb.TaskQueue{Name: s.taskQueue},
+					Memo: &commonpb.Memo{
+						Fields: map[string]*commonpb.Payload{"wfmemo1": wfMemo},
+					},
+					SearchAttributes: &commonpb.SearchAttributes{
+						IndexedFields: map[string]*commonpb.Payload{csa: wfSAValue},
+					},
+				},
+			},
+		},
+	}
+	req := &workflowservice.CreateScheduleRequest{
+		Namespace:  s.namespace,
+		ScheduleId: sid,
+		Schedule:   schedule,
+		Identity:   "test",
+		RequestId:  uuid.New(),
+		Memo: &commonpb.Memo{
+			Fields: map[string]*commonpb.Payload{"schedmemo1": schMemo},
+		},
+		SearchAttributes: &commonpb.SearchAttributes{
+			IndexedFields: map[string]*commonpb.Payload{csa: schSAValue},
+		},
+	}
+
+	var runs, runs2 int32
+	workflowFn := func(ctx workflow.Context) error {
+		workflow.SideEffect(ctx, func(ctx workflow.Context) any {
+			atomic.AddInt32(&runs, 1)
+			return 0
+		})
+		return nil
+	}
+	s.worker.RegisterWorkflowWithOptions(workflowFn, workflow.RegisterOptions{Name: wt})
+	workflow2Fn := func(ctx workflow.Context) error {
+		workflow.SideEffect(ctx, func(ctx workflow.Context) any {
+			atomic.AddInt32(&runs2, 1)
+			return 0
+		})
+		return nil
+	}
+	s.worker.RegisterWorkflowWithOptions(workflow2Fn, workflow.RegisterOptions{Name: wt2})
+
+	// create
+
+	createTime := time.Now()
+	_, err := s.engine.CreateSchedule(NewContext(), req)
```

Comment:
errr, the engine is actually the FrontendClient, it should be called frontendClient.

Reply:
no, not in this PR.